python/tvm/contrib/pyfrontend.py

```diff
@@ -0,0 +1,305 @@
+        if len(node.iter.args) == 1:
+            low, high = _api.const(0), self.visit(node.iter.args[0])
+        elif len(node.iter.args) == 2:
+            low, high = self.visit(node.iter.args[0]), self.visit(node.iter.args[1])
+        ext = high if isinstance(low, _expr.IntImm) and low.value == 0 else high - low
+        _body = [self.visit(stmt) for stmt in node.body]
+        _body = _list_to_block(_body)
+        _for_type = _stmt.For.Serial
+        if var in self.annotation.keys():
+            _for_type = eval('_stmt.For.%s' % self.annotation[var])
+        res = _make.For(self.loop_levels[var], low, ext, _for_type, 0, _body)
+        self.loop_levels.pop(var)
+        return res
+
+def parse(func, args):
+    """Parse a subset of Python to HalideIR
+
+    Parameters
+    ----------
+    func : str or types.FunctionType
+        If it is a string, parse the source code.
+        If it is a function, parse the function.
+
+    args : list of Buffer or Tensor or Var
+        The argument lists to the function.
+        Leave it None if no buffer is related to the function to be parsed
+
+    dump : bool, optional
+        If it is true, Python ast will be dumped.
+        A debug parameter.
+
+    Returns
+    -------
+    ir : Stmt
+        The result Halide IR
+    """
+    if isinstance(func, str):
+        ir = ast.parse(func)
+    else:
+        assert isinstance(func, types.FunctionType)
+        ir = ast.parse(inspect.getsource(func))
+    var_usage = _find_all_variable_decl(ir)
+    return PyAST2HalideIR(args, var_usage).visit(ir)
+
+def lower(func, args, binds=None):
+    """Lower a subset of Python to LoweredFunc
+
+    Parameters
+    ----------
+    func : str or types.FunctionType
+        If it is a string, parse the source code.
+        If it is a function, parse the function.
+
+    args : list of Buffer or Tensor or Var
+        The argument lists to the function.
+        Leave it None if no buffer is related to the function to be parsed
+
+    binds : dict of :any:`Tensor` to :any:`Buffer`, optional
+        Dictionary that maps the Tensor to Buffer which specified the data layout
+        requirement of the function. By default, a new compact buffer is created
+        for each tensor in the argument.
+
+    Returns
+    -------
+    ir : LoweredFunc
+        The result lowered function
+    """
+    if isinstance(func, str):
+        ir = ast.parse(func)
+    else:
+        assert isinstance(func, types.FunctionType)
+        ir = ast.parse(inspect.getsource(func))
+    if binds is None:
+        binds = {}
+    var_usage = _find_all_variable_decl(ir)
+    parser = PyAST2HalideIR(args, var_usage)
+    stmt = parser.visit(ir)
+    binds_args, args = builder.get_binds(parser.args)
+    binds_vars, _ = builder.get_binds(parser.vars_buffer)
+    _binds = binds_args.copy()
+    _binds.update(binds_vars)
+    _binds.update(binds_vars)
+    _binds.update(binds)
+    stmt = _ir_pass.StorageFlatten(stmt, _binds, 64)
```

Comment:
did you try other passes?

Reply:
No. I only tried this pass so far. Later, when more functionalities are supported, I will put more passes. So far, this pass is already sufficient to glue LLVM JIT backend.